social-network-data-migration-cli/src/main/java/pl/edu/agh/socialnetworkdatamigration/cli/SocialNetworkDataMigrationCLI.java

```diff
@@ -1,88 +1,153 @@
 
 public class SocialNetworkDataMigrationCLI {
     public static void main(String[] args) throws Exception {
-        if (args.length == 2 && args[0].equals("--i"))
-            interactiveSQL(args);
-        else if (args.length == 2)
-            SQLmigration(args);
-        else if (args[0].equals("--csv"))
-            if (args[1].equals("--i"))
-                interactiveCSV(args);
-            else
-                CSVmigration(args);
-        else
-            printUsage();
-    }
+        CommandLine cmd = getArgs(args);
+
+        Properties properties = new Properties();
+        properties.load(new FileInputStream(cmd.getOptionValue("config")));
+        String neo4jHost = properties.getProperty("neo4jHost");
+        String neo4jUser = properties.getProperty("neo4jUser");
+        String neo4jPassword = properties.getProperty("neo4jPassword");
+
+        if (cmd.hasOption("csv")) {
+            String dataPath = cmd.getOptionValue("csv");
+            String fieldTerminator = cmd.hasOption("field-terminator") ? cmd.getOptionValue("field-terminator") : "\\t";
+            boolean hasHeaders = !cmd.hasOption("no-headers");
+            CSVSchemaMapping schemaMapping;
+            CsvStrategy migrationStrategy;
+
+            if (cmd.hasOption('i')) {
+                schemaMapping = new InteractiveCSVMappingCreator(dataPath, hasHeaders).createInteractively();
+            } else {
+                String mappingsPath = cmd.getOptionValue("mapping");
+                String jsonStr = Files.readString(Path.of(mappingsPath));
+                schemaMapping = new CSVMappingLoader(dataPath, hasHeaders).loadFromJson(jsonStr);
+            }
 
-    private static void SQLmigration(String[] args) throws Exception {
-        String configPath = args[0];
-        String mappingsPath = args[1];
+            if (cmd.hasOption("merge")) {
+                migrationStrategy = new CsvMergingStrategy(dataPath, fieldTerminator, hasHeaders);
+            } else {
+                migrationStrategy = new CsvAddingStrategy(dataPath, fieldTerminator, hasHeaders);
+            }
+            try (Driver neo4jDriver = GraphDatabase.driver("neo4j://" + neo4jHost, AuthTokens.basic(neo4jUser, neo4jPassword));
+                 Migrator<CSVSchemaMapping> migrator = new Migrator<>(neo4jDriver, migrationStrategy, cmd.hasOption("dry-run"))) {
+                migrator.migrateData(schemaMapping);
+            }
+        } else {
+            String postgresHost = properties.getProperty("postgresHost");
+            String postgresDB = properties.getProperty("postgresDB");
+            String postgresUser = properties.getProperty("postgresUser");
+            String postgresPassword = properties.getProperty("postgresPassword");
+            SchemaMetaData schemaMetaData = new SchemaMetaData(postgresHost, postgresDB, postgresUser, postgresPassword);
+            SQLSchemaMapping schemaMapping;
+            PostgresStrategy migrationStrategy;
```

Comment:
W "Effective Java" piszą, że zmienne powinno się definiować jak najbliżej miejsca, w którym jest im przypisywana jakaś wartość. Tutaj przykładowo migrationStrategy można było zdefiniować trochę niżej. Nie mówię, żeby tracić czas na zmienianie tego, ale to taka mała sugestia na przyszłość :)